nuxeo-retention-web/elements/nuxeo-retention-analytics.js

```diff
@@ -0,0 +1,379 @@
+              options="[[_chartOptions()]]"
+            >
+            </chart-pie>
+          </nuxeo-card>
+        </div>
+
+        <!-- Expiration next 12 months -->
+        <nuxeo-repository-data
+          start-date="[[_formatDateNow()]]"
+          end-date="[[_formatDateFutureInMonths(12)]]"
+          with-date-intervals="month"
+          date-field="ecm:retainUntil"
+          data="{{expirationPerMonth}}"
+          index="[[index]]"
+        >
+        </nuxeo-repository-data>
+
+        <nuxeo-card class="chartBar" heading="[[i18n('retentionAnalytics.expirationNext12Months.heading')]]">
+          <chart-bar
+            labels="[[_futureExpirationLabels(expirationPerMonth)]]"
+            values="[[_values(expirationPerMonth)]]"
+            style="min-width:800px;"
+            options='{ "legend": { "display": false }, "animation": false }'
+          >
+          </chart-bar>
+        </nuxeo-card>
+      </nuxeo-page>
+    `;
+  }
+
+  static get is() {
+    return 'nuxeo-retention-analytics';
+  }
+
+  static get properties() {
+    return {
+      index: {
+        type: String,
+        value: '_all',
+      },
+      startDate: {
+        type: String,
+        value: moment().subtract(3, 'year').format('YYYY-MM-DD'),
+      },
+      endDate: {
+        type: String,
+        value: moment().format('YYYY-MM-DD'),
+      },
+    };
+  }
+
+  _values(data) {
+    return [
+      data.map((entry) => {
+        if (Array.isArray(entry.value)) {
+          return entry.value.map(this._values);
+        }
+        return entry.value;
+      }),
+    ];
+  }
+
+  _labels(data) {
+    return data.map((entry) => {
+      if (Array.isArray(entry.value)) {
+        return entry.value.map(this._labels);
+      }
+      return entry.key;
+    });
+  }
+
+  _extendEndDate(date) {
+    if (date && moment) {
+      return this._formatDate(moment(date).add(1, 'days').subtract(1, 'ms').toJSON());
+    }
+    return date;
+  }
+
+  _formatDateLabels(data, format) {
+    return data.map((entry) => {
+      if (Array.isArray(entry.value)) {
+        return entry.value.map(this._formatDateLabels);
+      }
+      return moment(entry.key).format(format);
+    });
+  }
+
+  _formatLabels(data, prefix) {
+    return data.map((entry) => {
+      if (Array.isArray(entry.value)) {
+        entry.value.map(this._formatLabels);
+      }
+      return this.i18n((prefix && prefix + entry.key) || entry.key);
+    });
+  }
+
+  _chartOptions() {
+    return {
+      legend: {
+        display: true,
+        position: 'right',
+        labels: {
+          boxWidth: 25,
+          fontSize: 12,
+        },
+      },
+      animation: {
+        duration: 100,
+        easing: 'easeOutQuart',
+        onComplete: this._drawValuesInPie,
+      },
+    };
+  }
+
+  // Source: https://stackoverflow.com/questions/33363373/how-to-display-pie-chart-data-values-of-each-slice-in-chart-js
+  // (to draw values inside the pie charts)
+  _drawValuesInPie() {
```

Comment:
I don't know if we should keep it, from my understanding, the main issue is the fact that, `ChartJS` doesn't provide by default the ability to add the labels on the chart pie. we should recheck the documentation again, also it seems there is another package based on char js, that allows this kind if things. The link was posted on the ⬆️ https://github.com/emn178/chartjs-plugin-labels. Also, I think we will need insight from UI team

Reply:
indeed, the only thing this code is doing is an animation but we should be fine without it. will investigate a little bit more.

Reply:
I was able to replace the code - the labels are still rendering fine.